packages/profile-sync-controller/src/controllers/user-storage/services.ts

```diff
@@ -136,21 +136,23 @@ export async function getUserStorageAllFeatureEntries(
       return null;
     }
 
-    const decryptedData = userStorage?.flatMap((entry) => {
+    const decryptedData: string[] = [];
+
+    for await (const entry of userStorage) {
       if (!entry.Data) {
-        return [];
+        continue;
       }
 
-      return encryption.decryptString(
-        entry.Data,
-        opts.storageKey,
-        nativeScryptCrypto,
+      decryptedData.push(
+        await encryption.decryptString(
+          entry.Data,
+          opts.storageKey,
+          nativeScryptCrypto,
+        ),
       );
-    });
+    }
 
-    return (await Promise.allSettled(decryptedData))
-      .map((d) => (d.status === 'fulfilled' ? d.value : undefined))
-      .filter((d): d is string => d !== undefined);
```

Comment:
Only difference we have in the new approach is that it doesn't handle failures and filter them out.

Hmm do we want to keep it in case there is a undecryptable value?

```ts
try {
  const data = await encryption.decryptString(
    entry.Data,
    opts.storageKey,
    nativeScryptCrypto,
  )
  decryptedData.push(data)
} catch {
  // do nothing
}
```